pkg/cmd/server/origin/run_components.go

```diff
@@ -151,6 +152,13 @@ func (c *MasterConfig) RunDNSServer() {
 	case "tcp6":
 		config.BindNetwork = "ipv6"
 	}
+	if c.Options.DNSConfig.ClusterDomain != "" {
+		if !strings.HasSuffix(c.Options.DNSConfig.ClusterDomain, ".") {
+			c.Options.DNSConfig.ClusterDomain += "."
```

Comment:
don't modify the `c.Options.DNSConfig` here, just build a local var with a trailing `.` you will use to set `config` fields